src/vote-subjects/vote-subjects.service.ts

```diff
@@ -1,39 +1,21 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { VoteSubject } from './entities/vote-subject.entity';
 
 @Injectable()
 export class VoteSubjectsService {
-  async findAll() {
-    return dummy;
+  constructor(
+    @InjectRepository(VoteSubject)
+    private voteSubjectRepository: Repository<VoteSubject>,
+  ) {
+    this.voteSubjectRepository = voteSubjectRepository;
+  }
+  async findAll(): Promise<VoteSubject[]> {
+    return this.voteSubjectRepository.find();
   }
 
-  async findById(id: string) {
-    return dummy.find((item) => item.id === id) ?? null;
+  async findById(id: string): Promise<VoteSubject | undefined> {
+    return this.voteSubjectRepository.findOne(id);
   }
```

Comment:
@sua-no 기존 로직 및 타입에 맞춰서 `null` 리턴이 좋을 것 같습니다~!

Reply:
null을 리턴하라고 하시는거는 아직 필요하지 않아서 말씀하신 부분인가요?

Reply:
`undefined` 대신 `null`이 좋을 것 같다는 말이었습니다~!

Reply:
undefined로 명시한게 repository에서 리턴하는 타입이 VoteSubject | undefined으로 명시되어서 인데 조건식으로 null을 리턴해도 에러가 발생하더라구요 undefined로 하면 문제가 될까요?

Reply:
@sua-no 문제 없습니다~! 리턴 타입을 null로 제안드렸다고 봐주시면 됩니다.

Reply:
```suggestion
 async findById(id: string): Promise<VoteSubject | null> {
    return this.voteSubjectRepository.findOne(id) ?? null;
}
```

Reply:
@sua-no 먼저 싱크를 맞추어야 할 부분은 Service 레이어에서 `findById` 함수 호출 결과가 해당 리소스가 없을 때, `undefined`와 `null`중에 무엇이 적합한지 맞추는 것일 것 같습니다.
`repository`에서 리턴하는 값은 컨트롤 가능한 부분이고 `subjectService.findById()`를 호출하는 인터페이스가 무엇을 반환하는지 적합하게 설계하는 것이 더 중요한 부분이라고 생각합니다.

Reply:
무슨말씀이신지 이해는 했는데  혹시 코드 받아서 해보셨을까요? 말씀해주신대로 해봤었는데 저렇게 작성했을 때 오류가 나서요

Reply:
```suggestion
async findById(id: string): Promise<VoteSubject | null> {
    const result = await this.voteSubjectRepository.findOne(id);
    return result ?? null;
}
```
@sua-no 혹시 타입 에러 말씀하시는 거면 이렇게 해보시죠...

Reply:
@hoontae24 아 제가 async/await 빼고 바로 return하는거로 작성했었네요ㅜㅜ 수정했습니다!